impl/src/main/java/org/ehcache/impl/internal/store/heap/SimpleMapWrapper.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.impl.internal.store.heap;
+
+import org.ehcache.config.EvictionVeto;
+import org.ehcache.core.spi.cache.Store;
+import org.ehcache.function.BiFunction;
+import org.ehcache.impl.internal.concurrent.ConcurrentHashMap;
+import org.ehcache.impl.internal.store.heap.holders.OnHeapValueHolder;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Random;
+
+/**
+ * Simple passthrough map wrapper, no key translation
+ */
+class SimpleMapWrapper<K, V> implements MapWrapper<K, V> {
```

Comment:
I thought about it but I am not too fond of the idea mostly because it pushes that code further away from its original source.
And also because the current decoupling is a nice place where to fix racy clear when using byte size heap